network_test.go

```diff
@@ -185,3 +185,31 @@ func TestUpdateRoutes(t *testing.T) {
 	assert.True(t, reflect.DeepEqual(results.Routes[0], testRoutes.Routes[1]),
 		"Interface created didn't match: got %+v, expecting %+v", results.Routes[0], testRoutes.Routes[1])
 }
+
+func TestListInterfaces(t *testing.T) {
```

Comment:
Those test are just checking the underlying network. How do you know something is going to be there? We should have stronger tests as I suggested earlier, by creating a netns and by creating some fake interfaces in it. Then, we can run the function listInterfaces() and it should find exactly what has been created.